web/src/components/njwds-extended/FileInput.tsx

```diff
@@ -0,0 +1,144 @@
+  accept?: AcceptedFileType[];
+  className?: string;
+  errorMessage?: string;
+  hasError?: boolean;
+  helperText: string;
+  maxSizeInMegabytes?: number; // filesize in bytes (1,048,576 = 1 MB)
+  value: InputFile | undefined;
+  onChange: (file: InputFile) => void;
+}
+
+const FileTypeToInputString: Record<AcceptedFileType, string> = {
+  PDF: "application/pdf",
+  PNG: "image/png",
+};
+
+const InputStringToFileType: Record<string, AcceptedFileType> = (
+  Object.keys(FileTypeToInputString) as AcceptedFileType[]
+).reduce((acc, key) => {
+  const value = FileTypeToInputString[key];
+  acc[value] = key;
+  return acc;
+}, {} as Record<string, AcceptedFileType>);
+
+export const FileInput = (props: Props): ReactElement => {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const fileInput = require("../../../../node_modules/@uswds/uswds/packages/usa-file-input/src");
+  const fileInputRef = useRef<HTMLInputElement>(null);
+  const BYTES_IN_A_MB = 1048576;
+  const size = props.maxSizeInMegabytes ? props.maxSizeInMegabytes * BYTES_IN_A_MB : undefined;
+
+  const [hasFileBeenRead, setHasFileBeenRead] = useState<boolean>(false);
+  const [fileSizeError, setFileSizeError] = useState({ hasError: false, errorMessage: "" });
+
+  const createAcceptedFileString = (): string | undefined => {
+    if (!props.accept) return undefined;
+    return props.accept.map((it) => FileTypeToInputString[it]).join(",");
+  };
+
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    if (e.target.files && e.target.files.length > 0) {
+      const uploadedFile = e.target.files[0];
+      const reader = new FileReader();
+      // eslint-disable-next-line unicorn/prefer-add-event-listener
+      reader.onload = function (event: ProgressEvent<FileReader>): void {
+        if (!event.target?.result) {
+          return;
+        }
+
+        setHasFileBeenRead(true);
+        props.onChange({
+          base64Contents: (event.target.result as string).split(",")[1],
+          sizeInBytes: uploadedFile.size,
+          fileType: InputStringToFileType[uploadedFile.type],
+          filename: uploadedFile.name,
+        });
+        setFileSizeError({ hasError: false, errorMessage: "" });
+      };
+      reader.readAsDataURL(uploadedFile);
+    }
+  };
+
+  useLayoutEffect(() => {
+    const fileInputElement = fileInputRef.current;
+
+    // Prevent the user from selecting a file that is too large or of the wrong file type
+    const preventInvalidFileSelection = (e: Event): void => {
+      const eventTarget = e.target as HTMLInputElement;
+      if (eventTarget.files && eventTarget.files.length > 0) {
+        const selectedFile = eventTarget.files[0];
+        if (props.accept && !props.accept.includes(InputStringToFileType[selectedFile.type])) {
+          e.stopImmediatePropagation();
+          setFileSizeError({
+            hasError: true,
+            errorMessage: `${selectedFile.name} file format is not supported.`,
```

Comment:
I think this error message (and the one below it) should be externalized into the config files. They can still use the `${}` string interpolation. There's a helper called `templateEval` that allows us to do that with config strings